docs/rfcs/032-compute-aux-files.md

```diff
@@ -0,0 +1,112 @@
+
+## Summary
+
+Files associated with replication slots differ from others in two main aspects:
+1. Durability Requirement: The durability requirement for replication slot files is higher. Losing a replication slot file means it cannot be regenerated, and replication will consequently break
+2. Access Pattern: Access to these files occurs while Postgres is running, not just at startup and shutdown.
+
+For statistics files, the stakes for durability are lower since these files can be regenerated, albeit with a performance cost.
+
+For all of that files Point-In-Time Recovery (PITR) is not strictly necessary. It would be a nice-to-have feature but is not essential.
+
+## Possible storages
+
+### s3fs
+
+Mounting an `/<branch_id>/<endpoint_id>` directory into each running compute node is a viable option.
+
+Pros:
+
+* Eliminates the need to modify Postgres code with special file access routines for WAL logging on write or some API access on read/write.
+* The file system API provides built-in access laziness, meaning prewarm, stats, and PGSS files don't need to be downloaded before database startup (background workers can read them as needed).
+
+Cons:
+
+* Mount time adds to start latency. Some slot files must be downloaded for Postgres to start, requiring careful batch access to prevent slow starts with thousands of small files. TODO: A detailed list of files that can block start needs to be compiled. TODO: check if snapshot files need to be read at startup.
+
+It not clear how to exactly mount S3 in our setup. We should restrict access to only a certain prefix, e.g. `/<branch_id>/<endpoint_id>`. Then we have to mount it somehow. AWS has `s3-csi`, but it is mounted on pod start and with our approach of pre-created computes we don't know endpoint/tenant during pod start. So we have to manage mount inside pod. That means setting `cap_admin` on pod. So not sure how to do that with pod. On the othe hand things are easier with NeonVM: in each pod we have VM and we have root access to it, so we can mount S3.
+
+TODO: Determine how to share credentials with the VM and restrict access to a specific prefix.
+
+Note: S3FS is not suited for general-purpose file systems due to its limitations, such as no in-place updates and high latency. However, it is suitable for the files discussed here.
+
+### EBS
+
+EBS is per-Availability Zone (AZ), introducing a dependency on AZ for branch/endpoint, which is undesirable.
+
+### EFS
+
+Built on top of NFS, EFS could be a potential solution but shares similar mounting issues with S3FS. Restricting access to specific paths is more challenging, and past issues with Vector on EFS leading to corrupted files (possibly due to Vector) make it a less favorable option.
```

Comment:
Using my proposal using tenant isolation with a per tenant ecryptfs on top of the underlying filesystem a single mount per pod would suffice.

Reply:
In addition to eCryptfs we should also evaluate goCryptfs because it has a friendlier MIT license (eCryptfs is GPL) and is simpler to use (user space instead of kernel space).